moveit_experimental/jog_arm/include/jog_arm/jog_arm_server.h

```diff
@@ -0,0 +1,284 @@
+       (-2. * filter_coeff_ * filter_coeff_ + 2.) * previous_filtered_measurements_[0]);
+
+  // Store the new filtered measurement
+  previous_filtered_measurements_[1] = previous_filtered_measurements_[0];
+  previous_filtered_measurements_[0] = new_filtered_msrmt;
+
+  return new_filtered_msrmt;
+}
+
+/**
+ * Class JogCalcs - Perform the Jacobian calculations.
+ */
+class JogCalcs
+{
+public:
+  JogCalcs(const JogArmParameters& parameters, jog_arm_shared& shared_variables,
+           const std::unique_ptr<robot_model_loader::RobotModelLoader>& model_loader_ptr);
+
+protected:
+  ros::NodeHandle nh_;
+
+  moveit::planning_interface::MoveGroupInterface move_group_;
+
+  sensor_msgs::JointState incoming_jts_;
+
+  bool cartesianJogCalcs(geometry_msgs::TwistStamped& cmd, jog_arm_shared& shared_variables);
+
+  bool jointJogCalcs(const moveit_experimental::JogJoint& cmd, jog_arm_shared& shared_variables);
+
+  // Parse the incoming joint msg for the joints of our MoveGroup
+  bool updateJoints();
+
+  Eigen::VectorXd scaleCartesianCommand(const geometry_msgs::TwistStamped& command) const;
+
+  Eigen::VectorXd scaleJointCommand(const moveit_experimental::JogJoint& command) const;
+
+  Eigen::MatrixXd pseudoInverse(const Eigen::MatrixXd& J) const;
+
+  bool addJointIncrements(sensor_msgs::JointState& output, const Eigen::VectorXd& increments) const;
+
+  // Reset the data stored in low-pass filters so the trajectory won't jump when
+  // jogging is resumed.
+  void resetVelocityFilters();
+
+  // Avoid a singularity or other issue.
+  // Needs to be handled differently for position vs. velocity control
+  void halt(trajectory_msgs::JointTrajectory& jt_traj);
+
+  void publishWarning(bool active) const;
+
+  bool checkIfJointsWithinBounds(trajectory_msgs::JointTrajectory_<std::allocator<void>>& new_jt_traj);
+
+  // Possibly calculate a velocity scaling factor, due to proximity of
+  // singularity and direction of motion
+  double decelerateForSingularity(Eigen::MatrixXd jacobian, const Eigen::VectorXd commanded_velocity);
+
+  // Apply velocity scaling for proximity of collisions and singularities
+  bool applyVelocityScaling(jog_arm_shared& shared_variables, trajectory_msgs::JointTrajectory& new_jt_traj,
+                            const Eigen::VectorXd& delta_theta, double singularity_scale);
+
+  trajectory_msgs::JointTrajectory composeOutgoingMessage(sensor_msgs::JointState& joint_state,
+                                                          const ros::Time& stamp) const;
+
+  void lowPassFilterVelocities(const Eigen::VectorXd& joint_vel);
+
+  void lowPassFilterPositions();
+
+  void insertRedundantPointsIntoTrajectory(trajectory_msgs::JointTrajectory& trajectory, int count) const;
+
+  const robot_state::JointModelGroup* joint_model_group_;
+
+  robot_state::RobotStatePtr kinematic_state_;
+
+  sensor_msgs::JointState jt_state_, original_jts_;
```

Comment:
variable names are inconsistent, seems the second one should be ``original_jt_state_``